react/features/polls/components/PollDialog.js

```diff
@@ -0,0 +1,363 @@
+            label: 'polls.results',
+            props: {},
+            submit: null
+        });
+
+        return (
+            <DialogWithTabs
+                closeDialog = { this._closeDialog }
+                cssClassName = { 'polls-dialog' }
+                key = { isPollRunning.toString() }
+                okTitleKey = { title }
+                onSubmit = { this._onSubmit }
+                tabs = { tabs }
+                titleKey = { 'dialog.polls' } />
+        );
+    }
+
+    _onSubmit: () => void;
+
+    /**
+     * Submit button handler.
+     *
+     * @returns {void}
+     */
+    _onSubmit() {
+        const { isPollRunning } = this.props;
+
+        if (isPollRunning) {
+            this._endPoll();
+        } else {
+            this._createNewPoll();
+        }
+    }
+
+    _addChoice: (void) => void;
+
+    /**
+     * Add new item to the list.
+     *
+     * @returns {void}
+     */
+    _addChoice() {
+        const id = v1();
+
+        this.setState({
+            choices: {
+                ...this.state.choices,
+                [id]: {
+                    id,
+                    text: '',
+                    votes: []
+                }
+            }
+        });
+    }
+
+    _removeChoice: (number) => void;
+
+    /**
+     * Removes an item from list.
+     *
+     * @param {string} id - ID of item in list.
+     * @returns {void}
+     */
+    _removeChoice(id: string) {
+        const filteredChoices = Object.assign({}, this.state.choices);
+
+        delete filteredChoices[id];
+
+        this.setState({
+            choices: filteredChoices
+        });
+    }
+
+    _choiceTextChange: (string, string) => void;
+
+    /**
+     * Item text change handler.
+     *
+     * @param {string} id - ID of the item with change.
+     * @param {string} text - New text.
+     * @returns {void}
+     */
+    _choiceTextChange(id: string, text: string) {
+        const choice = this.state.choices[id];
+
+        this.setState({
+            choices: {
+                ...this.state.choices,
+                [id]: {
+                    ...choice,
+                    text
+                }
+            }
+        });
+    }
+
+    _createNewPoll: () => void;
+
+    /**
+     * Click handler for creating a new poll.
+     *
+     * @returns {void}
+     */
+    _createNewPoll() {
+        const { question, choices, poll } = this.state;
+        const { dispatch } = this.props;
+        const uniqueChoices = getUniquePollChoices(choices);
+
+        if (!question.text.trim() || Object.keys(uniqueChoices).length < 2) {
+            return false;
+        }
+
+        Object.keys(uniqueChoices).forEach(x => {
+            poll.choices.push(x);
+        });
+
+        const payload = {
+            poll,
+            question,
+            choices: uniqueChoices
+        };
+
+        dispatch(startPoll(payload));
+
+        // Clear Object
+        const newState = this._clearState();
```

Comment:
I find this to be an uncommon pattern, to clear state manually like this right away. I think the comment could be expanded upon to explain why you're clearing state.

Reply:
This is needed to refresh the list of choices and assign new IDs for all the required objects. The main problem is that PollDialog which contains the state remains alive during creating and ending the vote therefore the state used to create the first vote will still be the same when creating the second or the third